File: push/src/main/java/com/xiaomi/channel/commonutils/msa/MiuiIdManager.java
package com.xiaomi.channel.commonutils.msa;

import android.content.Context;
import android.provider.Settings;
import com.xiaomi.channel.commonutils.android.SystemUtils;
import com.xiaomi.channel.commonutils.logger.MyLog;
import java.lang.reflect.Method;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/msa/MiuiIdManager.class */
class MiuiIdManager implements IdManager {
    private static final String CLASS_NAME = "com.android.id.impl.IdProviderImpl";
    private static final String KEY_ALLOW_OAID_USED = "allow_oaid_used";
    private static final String METHOD_AAID = "getAAID";
    private static final String METHOD_OAID = "getOAID";
    private static final String METHOD_UDID = "getUDID";
    private static final String METHOD_VAID = "getVAID";
    private static final String MIUI_PLATFORM_PKG = "com.xiaomi.xmsf";
    private Class<?> mClass;
    private Context mContext;
    private Object mIdProivderImpl;
    private Method mGetUDID = null;
    private Method mGetOAID = null;
    private Method mGetVAID = null;
    private Method mGetAAID = null;

    public MiuiIdManager(Context context) {
        this.mContext = context;
        loadClass(context);
    }

    private String invokeMethod(Context context, Method method) {
        Object obj = this.mIdProivderImpl;
        if (obj == null || method == null) {
            return null;
        }
        try {
            Object objInvoke = method.invoke(obj, context);
            if (objInvoke != null) {
                return (String) objInvoke;
            }
            return null;
        } catch (Exception e) {
            MyLog.e("miui invoke error", e);
            return null;
        }
    }

    public static boolean isMiuiPhone(Context context) {
        return "com.xiaomi.xmsf".equals(context.getPackageName());
    }

    private void loadClass(Context context) {
        try {
            Class<?> clsLoadClass = SystemUtils.loadClass(context, CLASS_NAME);
            this.mClass = clsLoadClass;
            this.mIdProivderImpl = clsLoadClass.newInstance();
            this.mGetUDID = this.mClass.getMethod(METHOD_UDID, Context.class);
            this.mGetOAID = this.mClass.getMethod(METHOD_OAID, Context.class);
            this.mGetVAID = this.mClass.getMethod(METHOD_VAID, Context.class);
            this.mGetAAID = this.mClass.getMethod(METHOD_AAID, Context.class);
        } catch (Exception e) {
            MyLog.e("miui load class error", e);
        }
    }

    @Override // com.xiaomi.channel.commonutils.msa.IdManager
    public String getAAID() {
        return invokeMethod(this.mContext, this.mGetAAID);
    }

    @Override // com.xiaomi.channel.commonutils.msa.IdManager
    public String getOAID() {
        return invokeMethod(this.mContext, this.mGetOAID);
    }

    @Override // com.xiaomi.channel.commonutils.msa.IdManager
    public String getUDID() {
        return null;
    }

    @Override // com.xiaomi.channel.commonutils.msa.IdManager
    public String getVAID() {
        return invokeMethod(this.mContext, this.mGetVAID);
    }

    @Override // com.xiaomi.channel.commonutils.msa.IdManager
    public boolean isAllowOAID() {
        boolean z = false;
        try {
            if (Settings.Secure.getInt(this.mContext.getContentResolver(), KEY_ALLOW_OAID_USED, 1) == 1) {
                z = true;
            }
            return z;
        } catch (Exception e) {
            MyLog.w("miui is allow oaid error" + e);
            return false;
        }
    }

    @Override // com.xiaomi.channel.commonutils.msa.IdManager
    public boolean isSupported() {
        return (this.mClass == null || this.mIdProivderImpl == null) ? false : true;
    }
}

File: legacy/src/main/java/com/xiaomi/channel/commonutils/msa/MiuiIdManager.kt
package com.xiaomi.channel.commonutils.msa

import android.content.Context
import android.provider.Settings
import com.xiaomi.channel.commonutils.android.SystemUtils
import com.xiaomi.channel.commonutils.logger.MyLog
import java.lang.reflect.Method

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/channel/commonutils/msa/MiuiIdManager.java
 */
class MiuiIdManager(private val mContext: Context) : IdManager {
    private var mClass: Class<*>? = null
    private var mIdProivderImpl: Any? = null
    private var mGetUDID: Method? = null
    private var mGetOAID: Method? = null
    private var mGetVAID: Method? = null
    private var mGetAAID: Method? = null

    init {
        loadClass(mContext)
    }

    private fun invokeMethod(context: Context, method: Method?): String? {
        val impl = mIdProivderImpl
        if (impl == null || method == null) {
            return null
        }
        return try {
            method.invoke(impl, context) as? String
        } catch (e: Exception) {
            MyLog.e("miui invoke error", e)
            null
        }
    }

    private fun loadClass(context: Context) {
        try {
            val cls = SystemUtils.loadClass(context, CLASS_NAME)
            mClass = cls
            mIdProivderImpl = cls.getDeclaredConstructor().newInstance()
            mGetUDID = cls.getMethod(METHOD_UDID, Context::class.java)
            mGetOAID = cls.getMethod(METHOD_OAID, Context::class.java)
            mGetVAID = cls.getMethod(METHOD_VAID, Context::class.java)
            mGetAAID = cls.getMethod(METHOD_AAID, Context::class.java)
        } catch (e: Exception) {
            MyLog.e("miui load class error", e)
        }
    }

    override fun getAAID(): String? = invokeMethod(mContext, mGetAAID)

    override fun getOAID(): String? = invokeMethod(mContext, mGetOAID)

    override fun getUDID(): String? = null

    override fun getVAID(): String? = invokeMethod(mContext, mGetVAID)

    override fun isAllowOAID(): Boolean {
        return try {
            Settings.Secure.getInt(mContext.contentResolver, KEY_ALLOW_OAID_USED, 1) == 1
        } catch (e: Exception) {
            MyLog.w("miui is allow oaid error$e")
            false
        }
    }

    override fun isSupported(): Boolean = mClass != null && mIdProivderImpl != null

    companion object {
        private const val CLASS_NAME = "com.android.id.impl.IdProviderImpl"
        private const val KEY_ALLOW_OAID_USED = "allow_oaid_used"
        private const val METHOD_AAID = "getAAID"
        private const val METHOD_OAID = "getOAID"
        private const val METHOD_UDID = "getUDID"
        private const val METHOD_VAID = "getVAID"
        private const val MIUI_PLATFORM_PKG = "com.xiaomi.xmsf"

        @JvmStatic
        fun isMiuiPhone(context: Context): Boolean = MIUI_PLATFORM_PKG == context.packageName
    }
}
